prov/efa/src/efa_strerror.c

```diff
@@ -1,134 +1,91 @@
-		? UNKNOWN_ERR_STR
-		: errno_str[err - FI_EFA_ERRNO_OFFSET];
+	return EFA_ERRNO_IS_PROV(err)
+		? prov_errno_str[err - EFA_PROV_ERRNO_START]
+		: "Unknown error";
 }
 
 /**
- * @brief   Convert EFA error into a printable string
- * @details Given a non-negative EFA-specific error code, this function returns
- *          a pointer to a static string that corresponds to it. This text is
- *          suitable for printing in debug/warning/error messages.
- *          If an optional enriched error message string is provided, it will
- *          override the default message and be returned instead.
- * @param[in]  err           An EFA-specific error code
- * @param[in]  enriched_err  An enriched error message. It should be obtained
- *                           from a prior fi_cq_readerr call.
- * @return     An error message with helpful debugging information specific to
- *             the error code
+ * @brief Convert an EFA error code into a short, printable string
+ *
+ * Given a non-negative EFA-specific error code, this function returns a pointer
+ * to a null-terminated string that corresponds to it; suitable for
+ * interpolation in logging messages.
+ *
+ * @param[in]	err    An EFA-specific error code
+ * @return	Null-terminated string with static storage duration (caller does
+ *		not free).
  */
-const char *efa_strerror(enum efa_errno err, const char *enriched_err)
+const char *efa_strerror(enum efa_errno err)
 {
-	if (enriched_err) {
-		return enriched_err;
-	}
+	return EFA_ERRNO_IS_PROV(err)
+		? efa_prov_strerror(err)
+		: efa_io_comp_strerror(err);
+}
 
-	return err >= FI_EFA_ERRNO_OFFSET
-	        ? efa_errno_str(err)
-	        : efa_io_comp_status_str(err);
+/**
+ * @brief Report a long-form help message to the user
+ *
+ * Given a non-negative EFA-specific error code, this function prints a
+ * long-form help message. These help messages are provided as a best effort
+ * attempt at debugging known issues, especially for `EFA_IO_COMP_STATUS_*`
+ * errors from RDMA Core.
+ *
+ * @param[in]	err    An EFA-specific error code
+ *
+ * @todo As this function grows, we should consider reading these long-form
+ * messages from disk, instead of baking the strings in.
+ */
+void efa_show_help(enum efa_errno err) {
+	char *help = "";
+	switch (err) {
+	case EFA_IO_COMP_STATUS_REMOTE_ERROR_BAD_DEST_QPN:
+		help = "This error is detected remotely; "
+		"typically encountered when the peer process is no longer present";
+		break;
+	case EFA_IO_COMP_STATUS_LOCAL_ERROR_UNRESP_REMOTE:
+		help = "This error is detected locally; "
+		"The connection status is unknown or was never established via "
+		"handshake. This typically indicates one or more misconfigured "
+		"EC2 instances; most often due to incorrect inbound/outbound "
+		"security group rules. Refer to the public AWS documentation "
+		"for EFA for up-to-date configuration requirements. "
+		"This error can also be encountered when a peer process is no "
+		"longer present.";
```

Comment:
Could you also add a note about subnet constraint? An internal customer hit this error last week trying to communicate across subnets.